File: PA42.java
//Time Complexity : O(n^2)
//Space Complexity: O(1) //only auxillary space allocated

class Solution {
    public List<Integer> getRow(int rowIndex) {
        
        List<Integer> ls=new ArrayList<>();   
        int nprev=1;
        int oprev=1;
        ls.add(1);
        for(int i=1; i<=rowIndex;i++){
            for(int j=0;j<=i;j++){
                if(j==0 || j==i){
                    if(j==i){
                        ls.add(1);
                    }else{
                        ls.set(j,1);
                    }                    
                }
                else{
                    nprev=ls.get(j);
                    ls.set(j,oprev+ls.get(j));
                    oprev= nprev;
                }                
            }
        }
        return ls;
    }
}

File: PA_42.py
#LC 119 - Pascals Triangle II
#Time Complexity - O(k^2)
#Space Complexity - O(k)
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        def helper(path,idx,rowIndex):
            if idx == rowIndex:
                return path
            i = len(path)-1
            while i > 0:
                path[i] = path[i] + path[i-1]
                i = i - 1
            path.append(1)
            
            return helper(path,idx+1,rowIndex)
        return(helper([1],0,rowIndex))